hibernate-core/src/main/java/org/hibernate/cfg/AbstractPropertyHolder.java

```diff
@@ -134,7 +134,7 @@ protected ConverterDescriptor makeAttributeConverterDescriptor(AttributeConversi
 
 	@Override
 	public boolean isInIdClass() {
-		return isInIdClass != null ? isInIdClass : parent != null ? parent.isInIdClass() : false;
+		return isInIdClass != null ? isInIdClass : ( parent != null && parent.isInIdClass() );
```

Comment:
just my personal opinion but not sure this helps code readability

Reply:
FWIW I think using `&&` instead of a nested `?:` is a lot more readable. However, the parens are unnecessary. `?:` has a lower precedence than `&&`, so I would write:

```jav`
return isInIdClass != null ? isInIdClass : parent != null && parent.isInIdClass();
```

Reply:
what about using  an If statement instead?
```
if ( isInIdClass != null ) {
   return isInIdClass;
}
else if ( parent != null ) {
   return parent.isInIdClass();
}
else {
   return false;
}```

Reply:
> FWIW I think using `&&` instead of a nested `?:` is a lot more readable. However, the parens are unnecessary. `?:` has a lower precedence than `&&`, so I would write:
> 
> ```java
> return isInIdClass != null ? isInIdClass : parent != null && parent.isInIdClass();
> ```

Yeah, technically we don't need parenthesis. I used it to improve the 'perceived complexity' from human code reader. I agree with you generally speaking `&&` is better, but in this case I think Andrea's suggestion makes sense and I am gonna accept his suggestion.

Reply:
IMO as soon as the ternary becomes in any way "non-simple" we ought to use if/else.  So I agree with Andrea's suggestion as well.